src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -92,11 +108,41 @@ class EnumeratorCommandFinder
 
     static Loop HandlerCallbackFn(CommandId id, void * context)
     {
-        EnumeratorCommandFinder * self = static_cast<EnumeratorCommandFinder *>(context);
+        auto self = static_cast<EnumeratorCommandFinder *>(context);
         return self->HandlerCallback(id);
     }
 };
 
+std::optional<CommandId> EnumeratorCommandFinder::FindCommandId(Operation operation, const ConcreteCommandPath & path)
+{
+    mOperation = operation;
+    mTarget    = path.mCommandId;
+
+    CommandHandlerInterface * interface =
+        CommandHandlerInterfaceRegistry::Instance().GetCommandHandler(path.mEndpointId, path.mClusterId);
+
+    if (interface == nullptr)
+    {
+        return std::nullopt; // no data: no interface
+    }
+
+    CHIP_ERROR err = (interface->*mCallback)(path, HandlerCallbackFn, this);
+    if (err == CHIP_ERROR_NOT_IMPLEMENTED)
+    {
+        return std::nullopt; // no data provided by the interface
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        // Report the error here since we lose actual error. This generally should NOT be possible as CHI usually returns
```

Comment:
```suggestion
        // Report the error here since we lose actual error. This generally should NOT be possible as CommandHandlerInterface usually returns
```